### ISSUE TITLE
Support for Oauth?

### ISSUE THREAD

**Issue by Stunner** [nuclearace/SwiftDiscord#58]
Was wondering if there is any support for [Oauth2](https://discordapp.com/developers/docs/topics/oauth2) namely, [Implicit Grant](https://discordapp.com/developers/docs/topics/oauth2#implicit-grant), or ideally, [Authorization Code Grant](https://discordapp.com/developers/docs/topics/oauth2#authorization-code-grant)? 

If not: 

* Are there plans to support any of these? 
* Would you be open to me (eventually) adding in this support?

**Comment by nuclearace** [comment]
I think OAuth2 flow is generally outside the scope of a client lib. Once you have a JWT from an OAuth session you can use it in the client with the Bearer prefix though. And there are some enums for the different scopes.

I'm not too sure on this though, but generally OAuth stuff involves separate libraries.

**Comment by Stunner** [comment]
I'll look into this further, but I was thinking some additions to this library to make the whole sign in process with Oauth nicer. In the case for iOS, possibly provide a webview that is set to load the url required to begin the oath process. And possibly some hackery to set the redirect uri so the client may be able to intercept it and go from there. 

This might warrant creating a new library just focused on the iOS platform itself, however. But I thought it would be good to bring this topic up for discussion.

**Comment by nuclearace** [comment]
Yeah I don't really want to include iOS specific code in the lib... Mostly because iOS is kind of a pain to install atm, and there's a good chance it'll be break and stay that way for awhile before someone notices.

**Comment by nuclearace** [comment]
That being said, I would probably need to see an implementation in a PR before I know if it's too hacky or scary to try and include in the lib.

**Comment by Stunner** [comment]
Fair enough, thanks for the quick reply.
